sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -170,8 +170,8 @@ object JdbcUtils extends Logging {
       case LongType => Option(JdbcType("BIGINT", java.sql.Types.BIGINT))
       case DoubleType => Option(JdbcType("DOUBLE PRECISION", java.sql.Types.DOUBLE))
       case FloatType => Option(JdbcType("REAL", java.sql.Types.FLOAT))
-      case ShortType => Option(JdbcType("INTEGER", java.sql.Types.SMALLINT))
-      case ByteType => Option(JdbcType("BYTE", java.sql.Types.TINYINT))
+      case ShortType => Option(JdbcType("SMALLINT", java.sql.Types.SMALLINT))
```

Comment:
Just to check, these are pretty standard type names in DBs?

Reply:
Yes. When a DB does not support a particularity type or uses a different type then the respective dialect should be modified. e.g. Postgres does not support tinyint and you 'll see that in the PostfresDialect.scala ByteType also gets mapped to SMALLINT. Have added a sanity test in PostgresIntegrationSuite.scala.